crates/bvs-vault-cw20-tokenized/tests/integration_test.rs

```diff
@@ -0,0 +1,1388 @@
+        let allowance: cw20::AllowanceResponse = app
+            .wrap()
+            .query_wasm_smart::<cw20::AllowanceResponse>(vault.addr(), &query)
+            .unwrap();
+        assert_eq!(allowance.allowance, Uint128::new(25));
+    }
+
+    // We can query AllAllowance like normal cw20 token
+    {
+        let query = cw20_base::msg::QueryMsg::AllAllowances {
+            owner: staker.to_string(),
+            start_after: None,
+            limit: None,
+        };
+        let allowances: cw20::AllAllowancesResponse = app
+            .wrap()
+            .query_wasm_smart::<cw20::AllAllowancesResponse>(vault.addr(), &query)
+            .unwrap();
+        assert_eq!(allowances.allowances.len(), 1);
+        assert_eq!(
+            allowances.allowances[0].spender,
+            app.api().addr_make("spender").to_string()
+        );
+        assert_eq!(allowances.allowances[0].allowance, Uint128::new(25));
+    }
+
+    // We can query AllSpenderAllowances like normal cw20 token
+    {
+        let query = cw20_base::msg::QueryMsg::AllSpenderAllowances {
+            spender: app.api().addr_make("spender").to_string(),
+            start_after: None,
+            limit: None,
+        };
+        let allowances: cw20::AllSpenderAllowancesResponse = app
+            .wrap()
+            .query_wasm_smart::<cw20::AllSpenderAllowancesResponse>(vault.addr(), &query)
+            .unwrap();
+        assert_eq!(allowances.allowances.len(), 1);
+        assert_eq!(allowances.allowances[0].owner, staker.to_string());
+        assert_eq!(allowances.allowances[0].allowance, Uint128::new(25));
+    }
+
+    // We can query AllAccounts like normal cw20 token
+    {
+        let query = cw20_base::msg::QueryMsg::AllAccounts {
+            start_after: None,
+            limit: None,
+        };
+        let accounts: cw20::AllAccountsResponse = app
+            .wrap()
+            .query_wasm_smart::<cw20::AllAccountsResponse>(vault.addr(), &query)
+            .unwrap();
+        // we have 3 account
+        // staker, spender that has allowance and random guy that has balance
+        assert_eq!(accounts.accounts.len(), 3);
+    }
+}
+
+#[test]
+fn test_proper_contract_name_and_version() {
+    // we need to test this to make sure
+    // that cw20 base instantiation doesn't interfere with
+    // the contract name and version
+    // check the instantiate entry point in this vault for more details
+    let app = &mut App::default();
+    let TestContracts { vault, .. } = TestContracts::init(app);
+
+    let raw = app
+        .wrap()
+        .query_wasm_raw(vault.addr(), b"contract_info".to_vec())
+        .expect("failed to query contract_info");
+
+    let binary = &cosmwasm_std::Binary::from(raw.unwrap());
+
+    let contract_info: ContractVersion =
+        cosmwasm_std::from_json(binary).expect("invalid contract_info format");
+
+    assert_eq!(
+        contract_info.contract,
+        concat!("crates.io:", env!("CARGO_PKG_NAME"))
```

Comment:
```suggestion
        "crates.io:bvs-vault-cw20-tokenized"
```
It's always better to be explicit than the test that the underlying function does the "same thing".